main_test.go

```diff
@@ -208,6 +208,7 @@ func TestE2ETestProxyCreatesRequestMetricForEachRequest(t *testing.T) {
 
 	// make request to api and track start / end time of the request to
 	startTime := time.Now()
+	time.Sleep(1 * time.Second)
```

Comment:
what's the motivation for sleeping in these tests? i'm wondering if there is a way to make the tests more resilient?

if we need a sleep before every call to `waitForMetricsInWindow`, let's add it into that method instead of needing to remember to call it every time we define `startTime`. (include a comment before that call to sleep to explain why it is necessary)

Reply:
It was the problem on local test environment. Looks like the waitForMetricsInWindow tried to wait for metrics in particular time range. To make it work on local env, there was a problem with timing as we need some delay between startTime and actual metric write